libs/sdk-backend-tiger/src/convertors/toBackend/afm/FilterConverter.ts

```diff
@@ -145,49 +164,138 @@ export function convertMeasureValueFilter(filter: IMeasureValueFilter): MeasureV
                 // this is effectively the same behavior as in bear
                 from: Math.min(originalFrom, originalTo),
                 to: Math.max(originalFrom, originalTo),
                 treatNullValuesAs,
+                ...applyOnResultProp,
             },
         };
     }
 
     return null;
 }
 
-export function convertRankingFilter(filter: IRankingFilter): RankingFilter {
+export function convertRankingFilter(
+    computeRatioMeasureNumerators: Map<Identifier, IMeasure<IMeasureDefinition>>,
+    filter: IRankingFilter,
+): RankingFilter {
     const { measure, attributes, operator, value } = filter.rankingFilter;
     const dimensionalityProp = attributes
         ? { dimensionality: attributes.map(toRankingFilterDimensionalityIdentifier) }
         : {};
+    const [filteredMeasureId, applyOnResult] = getActualFilteredMeasureInfo(
+        computeRatioMeasureNumerators,
+        measure,
+    );
+    const applyOnResultProp = applyOnResult == undefined ? {} : { applyOnResult };
+
     return {
         rankingFilter: {
-            measures: [toMeasureValueFilterMeasureQualifier(measure)],
+            measures: [toMeasureValueFilterMeasureQualifier(filteredMeasureId)],
             ...dimensionalityProp,
             operator: operator as RankingFilterBodyOperatorEnum,
             value,
+            ...applyOnResultProp,
         },
     };
 }
 
-export function convertVisualizationObjectFilter(
-    filter: IMeasureFilter | IFilter,
-): FilterDefinition | FilterDefinitionForSimpleMeasure | null {
+/**
+ * Return the actual measure which should be used for measure based filtering, together with the info how the
+ * corresponding filter should be applied (`undefined` means default application).
+ */
+function getActualFilteredMeasureInfo(
+    computeRatioMeasureNumerators: ComputeRatioMeasureNumerators,
+    measure: ObjRefInScope,
+): [ObjRefInScope, boolean?] {
+    const updatedFilteredMeasure = isLocalIdRef(measure)
+        ? computeRatioMeasureNumerators.get(measure.localIdentifier)
+        : undefined;
+    return updatedFilteredMeasure == undefined
+        ? [measure, undefined]
+        : [localIdRef(updatedFilteredMeasure.measure.localIdentifier), false];
+}
+
+type ComputeRatioMeasureNumerators = Map<Identifier, IMeasure<IMeasureDefinition>>;
+
+/**
+ * Find those ratio measures which are used in some measure based filter. For each such measure create its updated
+ * version with ratio computation off and with new local identifier. These new auxiliary measures are meant to be used
+ * as actual base for their respective filters which should be applied on source data (instead of the result) so that
+ * both numerator and denominator of the ratio measure are actually filtered.
+ *
+ * Return mapping from the original local identifiers to new auxiliary measures.
+ */
+function determineComputeRatioMeasureNumerators(
+    afmMeasures: Array<IMeasure>,
+    filters: Array<IFilter>,
```

Comment:
small stuff: IMeasure[] and IFilter[] is ok here. we try to stick with `[]` unless linter complains that we should use Array. they are equivalent.

Reply:
noted and fixed